apps/core-app/src/pages/Proposals.tsx

```diff
@@ -1,43 +1,130 @@
 import { NewProposalList } from '../components/NewProposalList';
 import { FORM } from '../legos/form';
-import { useMemo } from 'react';
+import SearchInput from '../components/SearchInput';
+import FilterDropdown from '../components/FilterDropdown';
+import { BaseProposalCard } from '../components/BaseProposalCard';
+import useDebounce from '../utils/debounceHook';
+import { PROPOSAL_STATUS } from '@daohaus/common-utilities';
 
-const LeftCard = styled(Card)`
+const ActionsContainer = styled.div`
   width: 100%;
-  min-width: 54rem;
-  max-width: 64rem;
-  @media ${widthQuery.md} {
-    max-width: 100%;
-    min-width: 0;
-  }
+  display: flex;
+  justify-content: space-between;
+  margin-bottom: 3rem;
+`;
+
+const SearchFilterContainer = styled.div`
+  display: flex;
+  gap: 2.1rem;
 `;
 
 export const VALID_NEW_PROPOSALS = [FORM.SIGNAL, FORM.SHARE_SWAP];
 
 export function Proposals() {
-  const { proposals } = useProposals();
+  const {
+    proposals,
+    setProposalsPaging,
+    proposalsNextPaging,
+    setProposalsFilter,
+    setProposals,
+  } = useProposals();
+  const { dao } = useDao();
+  const [searchTerm, setSearchTerm] = useState<string | ''>('');
+  const [filter, setFilter] = useState<string | ''>('');
+
+  const debouncedSearchTerm = useDebounce<string>(searchTerm, 700);
 
   const newProposals = useMemo(() => {
+    console.log('proposals', proposals);
     return Object.keys(FORM).map((key) => FORM[key]);
-  }, []);
+  }, [proposals]);
+
+  useEffect(() => {
+    console.log('deboun', debouncedSearchTerm);
+
+    const filterQuery =
+      filter !== ''
+        ? statusFilter(
+            PROPOSAL_STATUS[filter],
+            Number(dao?.votingPeriod) + Number(dao?.gracePeriod)
+          )
+        : undefined;
+
+    if (debouncedSearchTerm && debouncedSearchTerm !== '') {
+      setProposals(undefined);
+      setProposalsFilter({
+        ...filterQuery,
+        title_contains_nocase: debouncedSearchTerm,
+      });
+      setProposalsPaging(defaultDaoData.proposalsPaging);
+    } else {
+      setProposalsFilter(filterQuery);
+      setProposalsPaging(defaultDaoData.proposalsPaging);
+      setProposals(undefined);
+    }
+    // TODO: I don't want to fire on these others!!
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [debouncedSearchTerm]);
+
+  const handleSearchTermChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setSearchTerm((prevState) =>
+      prevState === event.target.value ? '' : event.target.value
+    );
+  };
+
+  const toggleFilter = (event: MouseEvent<HTMLButtonElement>) => {
+    const searchQuery =
+      searchTerm !== '' ? { title_contains_nocase: searchTerm } : undefined;
+    setFilter((prevState) => {
+      if (prevState === event.currentTarget.value) {
+        setProposalsFilter(searchQuery);
+        setProposalsPaging(defaultDaoData.proposalsPaging);
+        setProposals(undefined);
+        return '';
```

Comment:
Curious what this function is returning values to,

Reply:
i think its reseting to default to no filtering -- ive seen it in a few places, maybe it could be pulled out into its own function?

Reply:
im having a hard time deciphering here - is `event` using the `debounceSearchTerm`?